contracts/adapters/linkswap/LinkswapStakingAdapter.sol

```diff
@@ -0,0 +1,200 @@
+    address internal constant LSLP_LINK_AZUKI_POOL = 0xa74Ef3faB9E94578c79e0077f6Bd572C9efc8733;
+    address internal constant LSLP_LINK_DOKI_POOL = 0x795BD26b99082E59478cfe8d9Cd207bb196808E4;
+
+    address internal constant GOVERNANCE = 0x75D1aA733920b14fC74c9F6e6faB7ac1EcE8482E;
+    address internal constant GOVERNANCE_FEES = 0x0389d755C1833C9b350d4E8B619Eae16deFc1CbA;
+
+    /**
+     * @return Amount of staked tokens / rewards earned after staking for a given account.
+     * @dev Implementation of ProtocolAdapter interface function.
+     */
+    function getBalance(address token, address account) external view override returns (uint256) {
+        if (token == YFL) {
+            uint256 totalRewards = 0;
+
+            totalRewards += ERC20(GOVERNANCE).balanceOf(account);
+            
+            totalRewards += StakingRewards(LSLP_YFL_LINK_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_YFL_WETH_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_DPI_LINK_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_MASQ_WETH_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_BUSD_LINK_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_GSWAP_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_CEL_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_YAX_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_CFI_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_USDC_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_USDT_POOL).earned(account,0);
+
+            return totalRewards;
+        } else if (token == CFI) {
+            return StakingRewards(LSLP_LINK_CFI_POOL).earned(account,1);
+        } else if (token == MASQ) {
+            return StakingRewards(LSLP_MASQ_WETH_POOL).earned(account,1);
+        } else if (token == DPI) {
+            return StakingRewards(LSLP_DPI_LINK_POOL).earned(account,1);
+        } else if (token == CEL) {
+            return StakingRewards(LSLP_LINK_CEL_POOL).earned(account,1);
+        } else if (token == YAX) {
+            return StakingRewards(LSLP_LINK_YAX_POOL).earned(account,1);
+        } else if (token == GSWAP) {
+            return StakingRewards(LSLP_LINK_GSWAP_POOL).earned(account,1);
+        } else if (token == AZUKI) {
+            uint256 totalRewards = 0;
+            totalRewards += StakingRewards(LSLP_LINK_AZUKI_POOL).earned(account,1);
+            totalRewards += StakingRewards(LSLP_LINK_DOKI_POOL).earned(account,1);
+            return totalRewards;
+        } else if (token == LSLP_YFL_WETH) {
+            uint256 total = 0;
+            total += ERC20(LSLP_YFL_WETH_POOL).balanceOf(account);
+            total += ERC20(GOVERNANCE_FEES).balanceOf(account)
```

Comment:
Adding `ERC20(GOVERNANCE_FEES).balanceOf(account)` to every token does not seem correct as you are counting this amount multiple times. What token amount is counted in this function?

Reply:
@cr-igor If you see [this contract](https://etherscan.io/address/0x0389d755C1833C9b350d4E8B619Eae16deFc1CbA) the multiple different LSLP tokens are accumulating as governance fee rewards. I now see why adding this line to every token is not correct, since it will not differentiate between each LSLP token when calculating the balance, but what I am trying to achieve is to calculate the total number of locked tokens for each LSLP (since these are locked until someone initiates a claim) which includes those generated as governance rewards. Does that make sense?

Reply:
so it should be `ERC20(LSLP_YFL_WETH).balanceOf(GOVERNANCE_FEES)`

Reply:
Oh okay thanks, I didn't realise it could be anything other than `account`!